### ISSUE TITLE
Remove jQuery dependency

### ISSUE THREAD

**Issue by jharding** [twitter/typeahead.js#1180]

**Comment by Andreyco** [comment]
Nice one, especially for Bloodhound.

**Comment by mntnoe** [comment]
I agree - currently I have to shim jQuery to be able to run Bloodhound in a worker thread.

**Comment by jslegers** [comment]
@Andreyco & @mntnoe :

You might want to check out https://github.com/jslegers/typeahead.js.

In that fork, I removed `jQuery` as a dependency of `Bloodhound`.

You still need `jQuery` for `Typeahead`, but you no longer need it if you only care about `Bloodhound`.

Note that I replaced `$.deferred` with a `Promise` based implementation. To allow `Bloodhound` to be used older browsers, I added a `Promise` polyfill as a dependency for `Bloodhound`.
